### ISSUE TITLE
Consider making reflectable a dev_dependency

### ISSUE THREAD

**Issue by sethladd** [PointyCastle/pointycastle#99]
If reflectable isn't required at runtime, can you move it to a dev_dependency?

We use pointycastle in Flutter (btw, thanks for the library! :) but PC pulls in reflectable which pulls in analyzer pinned to a very specific version.

Moving reflectable to a dev_dependency (assuming that works) frees up Flutter to move to a newer version of analyzer.

Thanks very much!

**Comment by sethladd** [comment]
cc @hixie

**Comment by stevenroose** [comment]
@sethladd Sadle, reflectable is required at runtime:
https://github.com/PointyCastle/pointycastle/blob/master/lib/src/registry/registry.dart#L7

The Registry class is used to easily construct combined and hybrid algorithms.

cc @eernstg, perhaps reflectable can loosen the specific version requirement?

**Comment by eernstg** [comment]
It sounds like you haven't upgraded to use reflectable version 0.5.3. Reflectable has had a tight analyzer version constraint for months (`analyzer: 0.27.1`), but we recently found a way to address the underlying problem, and this enabled reflectable 0.5.3 to use the much more flexible constraint `analyzer: ^0.27.2`.

**Comment by Hixie** [comment]
We do now use the version with the less tight constraint. Our bigger problem though is that we'd like to drop our runtime dependency on the analyzer package entirely, as part of our efforts to reduce our dependency footprint.

**Comment by eernstg** [comment]
That's an understandable goal! ;-) But it shouldn't have much to do with code that is generated by the reflectable transformer: The generated code does not need to depend on the analyzer.

The transformer itself depends on the analyzer (crucially, so you cannot expect to remove that dependency), but that would affect the footprint of the build process, not the resulting application.

**Comment by Hixie** [comment]
@stevenroose earlier was saying that the `reflectable` package is required at runtime, though, no?

**Comment by eernstg** [comment]
Yes, 'reflectable.dart' is required at run time, but it does not depend on the analyzer.

The library 'reflectable.dart' is in fact adjusted to import 'src/reflectable_transformer_based.dart' rather than 'src/reflectable_mirror_based.dart', such that it does not cause a dependency on 'dart:mirrors' and such that it will be able to use the generated mirrors instead.

**Comment by Hixie** [comment]
Ah, ok. I thought reflectable depended on analyzer. My bad.
